Server/ReasnAPI/ReasnAPI/Services/AddressService.cs

```diff
@@ -1,16 +1,86 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
 
 namespace ReasnAPI.Services {
     public class AddressService (ReasnContext context) {
         private readonly ReasnContext _context = context;
 
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+        public AddressDto? CreateAddress(AddressDto? addressDto) {
+            if (addressDto is null)
+                return null;
+
+            var address = new Address {
+                City = addressDto.City,
+                Country = addressDto.Country,
+                State = addressDto.State,
+                Street = addressDto.Street,
+                ZipCode = addressDto.ZipCode
+            };
+
+            _context.Addresses.Add(address);
+            _context.SaveChanges();
+
+            return addressDto;
+        }
+
+        public AddressDto? UpdateAddress(int addressId, AddressDto? addressDto) {
+            if (addressDto is null)
+                return null;
+
+            var address = _context.Addresses.FirstOrDefault(r => r.Id == addressId);
+
+            if (address is null)
+                return null;
+
+            address.City = addressDto.City;
+            address.Country = addressDto.Country;
+            address.Street = addressDto.Street;
+            address.State = addressDto.State;
+            address.ZipCode = addressDto.ZipCode;
+
+            _context.SaveChanges();
+
+            return MapToAddressDto(address);
+        }
+
+        public void DeleteAddress(int addressId) {
```

Comment:
In case of delete operation I would return the boolean in order to inform whether the deletion was successful.